website/docs/pipelines/promoting-applications.mdx

```diff
@@ -277,6 +280,66 @@ subjects:
 ```
 </details>
 
+At this stage, pipeline controller can read the promotion secrets, however given there are no deny semantics in [RBAC](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#role-and-clusterrole):
+
+>An RBAC Role or ClusterRole contains rules that represent a set of permissions. Permissions are purely additive (there are no "deny" rules).
+
+You would need to ensure that no other `rolebindings` or `clusterrolebndings` would allow read the promotion secret at any time.
+
+This could be achieved in various ways. For example, by leveraging [policy capabilities](../../policy/intro/)
+to detect existing and future creation of roles that would grant read secrets permissions.
+
+<details><summary>Expand to see example</summary>
+
+Once enabled [policy and policy library](../../policy/weave-policy-profile/) you have a set of available policies to
+leverage. The one we are interested here is called `Rbac Prohibit Verbs On Resources` to restrict
+the creation of roles that would allow access to the promotion credentials.
+
+```yaml
+apiVersion: pac.weave.works/v2beta2
+kind: Policy
+metadata:
+    name: weave.templates.rbac-prohibit-verbs-on-resources
+spec:
+    id: weave.templates.rbac-prohibit-verbs-on-resources
+    name: Rbac Prohibit Verbs On Resources
+```
+that we could provide values with
+```yaml
+parameters:
+ - name: resource
+   type: string
+   required: true
+   value: "secrets"
+ - name: verb
+   type: string
+   required: true
+   value: "get" # do the same for list and watch
```

Comment:
Can we do a couple of things here?
 1. Put the full version of this for users so that they can copy and paste.
 2. Check what happens when someone uses `*` for verbs

Reply:
Given policy library is enterprise feature, I cannot add it here.